service/src/test/java/com/carhouse/service/impl/CarSaleServiceImplTest.java

```diff
@@ -56,34 +57,70 @@ void getCarSale() {
     void getNonExistentCarSale() {
         int carSaleId = 10;
         when(carSaleDao.getCarSale(carSaleId)).thenThrow(EmptyResultDataAccessException.class);
-        assertThrows(EmptyResultDataAccessException.class, () -> carSaleService.getCarSale(carSaleId));
+        NotFoundException thrown = assertThrows(NotFoundException.class, () -> carSaleService.getCarSale(carSaleId));
+        assertTrue(thrown.getMessage().contains("there is not such car sale"));
+
     }
 
     @Test
     void addCarSale() {
+        Integer newCarSaleId = 5;
         CarSale carSale = new CarSale(7);
-        carSaleService.addCarSale(carSale);
+        when(carSaleDao.addCarSale(carSale)).thenReturn(newCarSaleId);
+        assertEquals(newCarSaleId, carSaleService.addCarSale(carSale));
         verify(carSaleDao, times(1)).addCarSale(carSale);
     }
 
     @Test
     void addCarSaleWithWrongReference() {
         CarSale carSale = new CarSale(7);
         when(carSaleDao.addCarSale(carSale)).thenThrow(DataIntegrityViolationException.class);
-        assertThrows(DataIntegrityViolationException.class, () -> carSaleDao.addCarSale(carSale));
+        WrongReferenceException thrown = assertThrows(WrongReferenceException.class,
+                () -> carSaleService.addCarSale(carSale));
+        assertTrue(thrown.getMessage().contains("there is wrong references in your car sale"));
     }
 
     @Test
-    void updateCarSale() {
+    void updateCarSale() throws NotFoundException {
         CarSale carSale = new CarSale(5);
-        carSaleService.updateCarSale(carSale);
+        when(carSaleDao.updateCarSale(carSale)).thenReturn(true);
+        boolean isUpdated = carSaleService.updateCarSale(carSale);
+        assertTrue(isUpdated);
         verify(carSaleDao, times(1)).updateCarSale(carSale);
     }
 
     @Test
-    void deleteCarSale() {
+    void updateCarSaleWithWrongReference() {
+        CarSale carSale = new CarSale(4);
+        when(carSaleDao.updateCarSale(carSale)).thenThrow(DataIntegrityViolationException.class);
+        WrongReferenceException thrown = assertThrows(WrongReferenceException.class,
+                () -> carSaleService.updateCarSale(carSale));
+        assertTrue(thrown.getMessage().contains("there is wrong references in your car sale"));
+    }
+
+    @Test
+    void updateNotExistCarSale() {
+        CarSale carSale = new CarSale(17);
+        when(carSaleDao.getCarSale(carSale.getCarSaleId())).thenThrow(EmptyResultDataAccessException.class);
+        NotFoundException thrown = assertThrows(NotFoundException.class, () -> carSaleService.updateCarSale(carSale));
+        assertTrue(thrown.getMessage().contains("there is not such car sale"));
```

Comment:
what car? Why is there absent `carId` from request?

Reply:
Ok, I've fixed it